### PR TITLE
[AAATAR-178] Mint with soul points

### PR DESCRIPTION
**Pull request by DidacSF**
## Description

See https://ajunanetwork.atlassian.net/browse/AAATAR-178 for details. In short:

* Added soul points attribute with calculation logic on Avatar generation.

## Type of changes

- [ ] `build`: Changes that affect the build system or external dependencies (eg, Cargo, Docker)
- [ ] `ci`: Changes to CI configuration
- [ ] `docs`: Changes to documentation only
- [x] `feat`: Changes to add a new feature
- [ ] `fix`: Changes to fix a bug
- [ ] `refactor`: Changes that do not alter functionality
- [ ] `style`: Changes to format the code
- [ ] `test`: Changes to add missing tests or correct existing tests

## Checklist

- [x] Tests for the changes have been added
- [x] Necessary documentation is added (if appropriate)
- [x] Formatted with `cargo fmt --all`
- [x] Linted with `cargo clippy --all-features --all-targets`
- [x] Tested with `cargo test --workspace --all-features --all-targets`

### CODE REVIEW

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 1]
```diff
@@ -411,6 +412,19 @@ pub mod pallet {
 			(random_tier, random_variation)
 		}
 
+		#[inline]
+		fn calculate_soul_points_from_dna(dna: &Dna, season: &SeasonOf<T>) -> SoulPoints {
+			// We meed to calculate the maximum possible value for a single DNA strand based on
```
Comment (raulmartinezm) [inline pallets/ajuna-awesome-avatars/src/lib.rs:417]:
```suggestion
			// We need to calculate the maximum possible value for a single DNA strand based on
```

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 2]
```diff
@@ -411,6 +412,19 @@ pub mod pallet {
 			(random_tier, random_variation)
 		}
 
+		#[inline]
+		fn calculate_soul_points_from_dna(dna: &Dna, season: &SeasonOf<T>) -> SoulPoints {
```
Comment (raulmartinezm) [inline pallets/ajuna-awesome-avatars/src/lib.rs:416]: I would pass just `max_variations` instead of `season`.

**Review by raulmartinezm — COMMENTED** [review]
LGTM, just a couple of nitpick comments

File: pallets/ajuna-awesome-avatars/src/lib.rs (hunk 2 above)
Comment (DidacSF) [reply pallets/ajuna-awesome-avatars/src/lib.rs:416]: I think it's more explicit like this, you don't want people passing wrong values by mistake, so forcing you to pass the reference to a season is more self-documenting.

File: pallets/ajuna-awesome-avatars/src/lib.rs (hunk 1 above)
Comment (DidacSF) [reply pallets/ajuna-awesome-avatars/src/lib.rs]: Fixed!

**Review by cowboy-bebug — APPROVED** [review]

**Pull request merged**